Document the WebCodecs processing model. (#25812)
mdn/content: files/en-us/web/api/webcodecs_api/index.md

The WebCodecs API provides access to codecs that are already in the browser.
It gives access to raw video frames, chunks of audio data, image decoders, audio and video encoders and decoders.

### Processing Model

The WebCodecs API uses an asynchronous [processing model](https://w3c.github.io/webcodecs/#codec-processing-model-section). Each instance

> of an encoder or decoder maintains an internal, independent processing queue. When queueing substantial amount of work, it's important to

Reviewer: ```suggestion
of an encoder or decoder maintains an internal, independent processing queue. When queueing a substantial amount of work, it's important to
```

keep this model in mind.

> Methods named `configure()`, `encode()`, `decode()`, and `flush()` operate asynchronously by appending control messages
> to the end the queue. While methods named `reset()` and `close()` synchronously abort all pending work and purge the
> processing queue. After `reset()` more work may be queued following a call to `configure()`, but `close()` is a permanent operation.

Reviewer 2: ```suggestion
Methods named `configure()`, `encode()`, `decode()`, and `flush()` operate asynchronously by appending control messages
to the end the queue, while methods named `reset()` and `close()` synchronously abort all pending work and purge the
processing queue. After `reset()`, more work may be queued following a call to `configure()`, but `close()` is a permanent operation.
```

> The WebCodecs API uses an asynchronous [processing model](https://w3c.github.io/webcodecs/#codec-processing-model-section). Each instance
> of an encoder or decoder maintains an internal, independent processing queue. When queueing a substantial amount of work, it's important to
> keep this model in mind.
> 
> Methods named `configure()`, `encode()`, `decode()`, and `flush()` operate asynchronously by appending control messages
> to the end the queue. While methods named `reset()` and `close()` synchronously abort all pending work and purge the
> processing queue. After `reset()` more work may be queued following a call to `configure()`, but `close()` is a permanent operation.
> 
> Methods named `flush()` can be used to wait for the completion of all work that was pending at the time `flush()` was called. However, it
> should generally only be called once all desired work is queued. It is not intended to force progress at regular intervals. Calling it
> unnecessarily will affect encoder quality and cause decoders to require the next input to be a key frame.
> 
> ### Demuxing
> 
> There is currently no API for demuxing media containers. Developers working with containerized media will need to implement their own

Reviewer: ```suggestion
The WebCodecs API uses an asynchronous [processing model](https://w3c.github.io/webcodecs/#codec-processing-model-section). Each instance
of an encoder or decoder maintains an internal, independent processing queue. When queueing a substantial amount of work, it's important to
keep this model in mind.

The `configure()`, `encode()`, `decode()`, and `flush()` methods operate asynchronously by appending control messages
to the end of the queue. The `reset()` and `close()` methods synchronously abort all pending work and purge the
processing queue. After the `reset()` method is called, more work may be queued following a call to the `configure()` method, but `close()` is a permanent operation.

The `flush()` methods can be used to wait for the completion of all work that was pending at the time `flush()` was called. However, it
should generally only be called once all desired work is queued. It is not intended to force progress at regular intervals. Calling it
unnecessarily will affect encoder quality and cause decoders to require the next input to be a key frame.

### Demuxing

There is currently no API for demuxing media containers. Developers working with containerized media will need to implement their own
```

Author: These changes are fine with me, I even had this version prior to upload. However, I wasn't sure about the correctness of using `_The_ $method_list` versus `Methods named $method_list` since not all interfaces have all methods. E.g., `decode()` only applies to decoders, `encode()` to encoders, and `flush()` isn't needed on ImageDecoder.

Reviewer: Looks good to me. Pinging @sideshowbarker for final review and merge.
